### PR TITLE
Make operator overloads bind in more cases

### PR DESCRIPTION
**Pull request by adayton1**

### CODE REVIEW

**Review by rhornung67 — APPROVED** [review]

File: src/chai/ManagedArray.inl [hunk 1]
```diff
@@ -613,7 +613,7 @@ template<typename T>
 CHAI_INLINE
 CHAI_HOST_DEVICE
 bool
-ManagedArray<T>::operator== (ManagedArray<T>& rhs) const
+ManagedArray<T>::operator== (const ManagedArray<T>& rhs) const
 {
   return (m_active_pointer ==  rhs.m_active_pointer);
```
Comment (adayton1) [inline src/chai/ManagedArray.inl:618]: I'm also wondering if this should be `m_pointer_record == rhs.m_pointer_record`. Or if the host case should pull the host pointer from the pointer record: `m_pointer_record->m_pointers[CPU] == rhs.m_pointer_record->m_pointers[CPU]`

**Review by robinson96 — APPROVED** [review]
LGTM

**Review by davidbeckingsale — APPROVED** [review]

**Comment by adayton1** [comment]
@davidbeckingsale, looks like the tests failed due to the spack build pointing at the 2022.03 versions of CHAI's dependencies.

**Comment by davidbeckingsale** [comment]
@adayton1 I merged this anyway since the CI is in flux at the moment.

**Comment by rhornung67** [comment]
According to Umpire GItLab CI settings page (and several other projects), corona runners have been offline for 4 days. I pings LC GitLab admins on mattermost

**Pull request merged**